crates/protocol/src/registry.rs

```diff
@@ -166,18 +166,18 @@ impl<'de> Deserialize<'de> for PackageId {
 
 #[derive(Debug, Clone, Hash, PartialEq, Eq, Serialize, Deserialize)]
 #[serde(transparent)]
-pub struct LogId(AnyHash);
+pub struct LogId(pub AnyHash);
 
 impl LogId {
     pub fn operator_log<D: SupportedDigest>() -> Self {
         let prefix: &[u8] = b"WARG-OPERATOR-LOG-ID-V0".as_slice();
-        let hash: Hash<D> = Hash::of(prefix);
+        let hash: Hash<D> = Hash::of(Hash::<D>::of(prefix));
         Self(hash.into())
     }
 
     pub fn package_log<D: SupportedDigest>(id: &PackageId) -> Self {
         let prefix: &[u8] = b"WARG-PACKAGE-LOG-ID-V0:".as_slice();
-        let hash: Hash<D> = Hash::of((prefix, id));
+        let hash: Hash<D> = Hash::of(Hash::<D>::of((prefix, id)));
```

Comment:
Before we merge this, can we document why this double hash is necessary here (and likewise for the operator log id)?

At first glance, a `LogId` functions simply as an identifier, meaning as long as the same identifier is derived from the same package identifier (and internal prefix), then I would expect a single hash to suffice.

If there is some expected way this interacts with the transparency data structures, it should be called out explicitly.

Reply:
I agree with Peter that this should not need to be hashed twice.

A package's `LogId` should be a simple hash of the prefix string and the name.